### PR TITLE
doc: add upgrade document for 3.2

### PR DESCRIPTION
**Pull request by agarwal-mudit**
Added upgrade document for 3.2 with a separate section on
CSI sidecar containers update.

Signed-off-by: Mudit Agarwal <muagarwa@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@agarwal-mudit CI failure

File: docs/ceph-csi-upgrade.md [hunk 1]
```diff
@@ -345,3 +352,20 @@ reboot.
...
+### CSI Sidecar containers consideration
+
+With 3.2.0 version of ceph-csi we have updated the versions of CSI sidecar
+containers. These versions are generally compatible with kubernetes
+version>=1.17 but based on the kubernetes version you are using, you need to
+update the templates with required sidecar versions.
+You also might need to update or remove a few arguments based on the sidecar
+versions you are using. Here is a table for your reference:
+
+| Sidecar container | Updated version | Minimum Kubernetes version |
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md:365]: let's add a link to https://kubernetes-csi.github.io/docs/sidecar-containers.html for the compatibility check. if we do that we don't need to maintain our own list here.

File: docs/ceph-csi-upgrade.md [hunk 2]
```diff
@@ -76,6 +78,11 @@ to upgrade from cephcsi v2.1 to v3.0
 
 ## Upgrading from v3.0 to v3.1
 
+Refer [upgrade-from-v3.0-v3.1](https://github.com/ceph/ceph-csi/blob/v3.1.0/docs/ceph-csi-upgrade.md)
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md:81]: the link should be https://github.com/ceph/ceph-csi/blob/v3.1.2/docs/ceph-csi-upgrade.md v3.1.2 is the latest release we have

File: docs/ceph-csi-upgrade.md (hunk 1 above)
Comment (agarwal-mudit) [reply docs/ceph-csi-upgrade.md]: fixed

File: docs/ceph-csi-upgrade.md (hunk 2 above)
Comment (agarwal-mudit) [reply docs/ceph-csi-upgrade.md]: fixed

File: docs/ceph-csi-upgrade.md [hunk 3]
```diff
@@ -24,6 +25,7 @@
         - [4.2 Update the RBD Nodeplugin daemonset](#42-update-the-rbd-nodeplugin-daemonset)
         - [4.3 Manual deletion of RBD Nodeplugin daemonset pods](#43-manual-deletion-of-rbd-nodeplugin-daemonset-pods)
     - [Handling node reboot hangs due to existing network mounts](#handling-node-reboot-hangs-due-to-existing-network-mounts)
+    - [CSI Sidecar containers considerations](#csi-sidecar-containers-considerations)
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md:28]: link is not working, need to be changed to `[CSI Sidecar containers consideration](#csi-sidecar-containers-consideration)`

File: docs/ceph-csi-upgrade.md [hunk 4]
```diff
@@ -345,3 +352,15 @@ reboot.
...
+
+### CSI Sidecar containers consideration
+
+With 3.2.0 version of ceph-csi we have updated the versions of CSI sidecar
+containers. These versions are generally compatible with kubernetes
+version>=1.17 but based on the kubernetes version you are using, you need to
+update the templates with required sidecar versions.
+You also might need to update or remove a few arguments based on the sidecar
+versions you are using.
+Refer
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md:364]: Refer [sidecar-compatibility](https://kubernetes-csi.github.io/docs/sidecar-containers.html) for mode details?

File: docs/ceph-csi-upgrade.md (hunk 3 above)
Comment (agarwal-mudit) [reply docs/ceph-csi-upgrade.md]: fixed

File: docs/ceph-csi-upgrade.md (hunk 4 above)
Comment (agarwal-mudit) [reply docs/ceph-csi-upgrade.md:364]: fixed

**Review by Madhu-1 — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Pull request merged**